File: leetCodeDaily/JavaScript/Array/566_prblm.js
var matrixReshape = function (mat, r, c) {
  let m = mat.length,
    n = mat[0].length;

  // If reshape is not possible, return the original matrix
  if (m * n !== r * c) return mat;

  // Create a new matrix of size r x c
  let reshaped = new Array(r).fill(0).map(() => new Array(c));

  // Fill new matrix using 1D index
  for (let i = 0; i < m * n; i++) {
    reshaped[Math.floor(i / c)][i % c] = mat[Math.floor(i / n)][i % n];
  }

  return reshaped;
};

// Driver code
let mat = [
  [1, 2],
  [3, 4],
];
let r = 1,
  c = 4;

console.log("Reshaped Matrix:", matrixReshape(mat, r, c));

File: leetCodeDaily/C++/Array/566_prblm.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<vector<int>> matrixReshape(vector<vector<int>> &mat, int r, int c)
{
  int m = mat.size(), n = mat[0].size();

  // If reshape is not possible, return original matrix
  if (m * n != r * c)
    return mat;

  // Create a new matrix with r x c dimensions
  vector<vector<int>> reshaped(r, vector<int>(c));

  // Fill the new matrix using a single index i
  for (int i = 0; i < m * n; i++)
  {
    reshaped[i / c][i % c] = mat[i / n][i % n];
  }

  return reshaped;
}

// Driver code
int main()
{
  vector<vector<int>> mat = {{1, 2}, {3, 4}};
  int r = 1, c = 4;

  vector<vector<int>> reshaped = matrixReshape(mat, r, c);

  cout << "Reshaped Matrix:\n";
  for (auto row : reshaped)
  {
    for (auto num : row)
    {
      cout << num << " ";
    }
    cout << endl;
  }
  return 0;
}
